Assignment4/Rulo401-190243/task07.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1tV5j-DRcpPtoJGoMj8v2DSqR_9wyXeiE
+
+**Task 07: Querying RDF(s)**
+"""
+
+!pip install rdflib 
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2020-2021/master/Assignment4"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/resources/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**"""
+
+from rdflib.plugins.sparql import prepareQuery
+ns = Namespace("http://somewhere#")
+
+for s,p,o in g.triples((None, RDFS.subClassOf, ns.Person)):
+  print(s)
+
+q1 = prepareQuery('''
+  SELECT ?Subclasses WHERE { 
+    ?Subclasses RDFS:subClassOf ns:Person. 
+  }
+  ''',
+  initNs = { "ns": ns, "RDFS": RDFS}
+)
+
+for r in g.query(q1):
+  print(r.Subclasses)
+
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**"""
+
+for s,p,o in g.triples((None, RDF.type, ns.Person)):
+  print(s)
+pSubclass = g.triples((None, RDFS.subClassOf, ns.Person))
+for s,p,o in pSubclass:
+  people = g.triples((None, RDF.type, s))
+  for s,p,o in people:
+    print(s)
+
+
+q2 = prepareQuery('''
+  SELECT DISTINCT ?Person WHERE { 
+    {?Person RDF:type ns:Person} UNION
+    {?Person RDF:type ?Class.
+    ?Class  RDFS:subClassOf ns:Person}
```

Comment:
you need to have ALL the subclasses. Hint you may use chained properties with "*" for 0..n repetitions